indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/ledger/IndyPoolLedgerUser.kt

```diff
@@ -0,0 +1,308 @@
+package com.luxoft.blockchainlab.hyperledger.indy.ledger
+
+import com.luxoft.blockchainlab.hyperledger.indy.models.*
+import com.luxoft.blockchainlab.hyperledger.indy.utils.SerializationUtils
+import mu.KotlinLogging
+import org.hyperledger.indy.sdk.ledger.Ledger
+import org.hyperledger.indy.sdk.pool.Pool
+import java.lang.Thread.sleep
+import java.util.*
+
+
+const val RETRY_DELAY_MS: Long = 100L
+const val RETRY_TIMES: Int = 10
+
+/**
+ * This is an implementation of [LedgerUser] which uses standard indy pool as ledger.
+ *
+ * @param pool [Pool] - indy pool handle
+ * @param did [String] - indy user did
+ * @param signProvider - lambda that is used to sign ledger requests
+ */
+class IndyPoolLedgerUser(val pool: Pool, val did: String, val signProvider: (data: String) -> String) : LedgerUser {
```

Comment:
Lets move did to interface?
And do did compression when we build IndyUser?

Reply:
Write lowercase chars instead of uppercase?

Reply:
I have some thoughts on this. Let's discuss face-to-face.